addons/purchase/models/account_invoice.py

```diff
@@ -55,6 +55,8 @@ def _onchange_purchase_auto_complete(self):
         new_currency_id = self.invoice_line_ids and self.currency_id or invoice_vals.get('currency_id')
         del invoice_vals['ref']
         del invoice_vals['company_id']  # avoid recomputing the currency
+        if 'move_type' in invoice_vals and self.move_type == invoice_vals['move_type']:
```

Comment:
move_type should always be in `invoice_vals`, since the original `_prepare_invoice` method sets it